syntax = "proto3";

service Logger {
  rpc WriteBubble (WriteBubbleArgs) returns (WriteBubbleReply) {}

  // Write log entry.
  rpc WriteLog (LogEntry) returns (Empty) {}

  rpc DumpSched (DumpSchedEntry) returns (Empty) {}

  rpc DumpStepSched (DumpStepSchedEntry) returns (Empty) {}

  rpc RecordInstr (RecordInstrEntry) returns (Empty) {}

  rpc RecordSideTask (RecordSideTaskEntry) returns (Empty) {}

  rpc Clear(Empty) returns (Empty) {}
}

message WriteBubbleArgs {
  double start = 1;
  double end = 2;
  int64 stage_id = 3;
  int64 global_rank = 4;
  string device = 5;
}

message WriteBubbleReply {

}

message LogEntry {
  int64 pid = 1;
  int64 ts = 2;
  string msg = 3;
}

message DumpSchedEntry {
  int64 pid = 1;
  int64 ts = 2;
  string msg = 3;
}

message DumpStepSchedEntry {
  int64 pid = 1;
  int64 ts0 = 2;
  int64 ts1 = 3;
  string msg = 4;
}

message RecordInstrEntry {
  int64 pid = 1;
  int64 ts0 = 2;
  int64 ts1 = 3;
  string instr = 4;
}

message RecordSideTaskEntry {
  int64 pid = 1;
  int64 ts0 = 2;
  int64 ts1 = 3;
  int64 counter = 4;
}

message Empty {

}